packages/block-library/src/query/deprecated.js

```diff
@@ -296,6 +296,82 @@ const v3 = {
 	},
 };
 
-const deprecated = [ v3, v2, v1 ];
+const v4 = {
+	attributes: {
+		queryId: {
+			type: 'number',
+		},
+		query: {
+			type: 'object',
+			default: {
+				perPage: null,
+				pages: 0,
+				offset: 0,
+				postType: 'post',
+				order: 'desc',
+				orderBy: 'date',
+				author: '',
+				search: '',
+				exclude: [],
+				sticky: '',
+				inherit: true,
+				taxQuery: null,
+				parents: [],
+			},
+		},
+		tagName: {
+			type: 'string',
+			default: 'div',
+		},
+		displayLayout: {
+			type: 'object',
+			default: {
+				type: 'list',
+			},
+		},
+		namespace: {
+			type: 'string',
+		},
+	},
+	supports: {
+		align: [ 'wide', 'full' ],
+		html: false,
+		color: {
+			gradients: true,
+			link: true,
+			__experimentalDefaultControls: {
+				background: true,
+				text: true,
+			},
+		},
+		__experimentalLayout: true,
+	},
+	save( { attributes: { tagName: Tag = 'div' } } ) {
+		const blockProps = useBlockProps.save();
+		const innerBlocksProps = useInnerBlocksProps.save( blockProps );
+		return <Tag { ...innerBlocksProps } />;
+	},
+	isEligible: ( { layout } ) =>
+		! layout ||
+		layout.inherit ||
+		( layout.contentSize && layout.type !== 'constrained' ),
+	migrate: ( attributes ) => {
+		const { layout = null } = attributes;
+		if ( ! layout ) {
+			return attributes;
+		}
+		if ( layout.inherit || layout.contentSize ) {
+			return {
+				...attributes,
+				layout: {
+					...layout,
```

Comment:
Just noticed that if `layout.inherit` is `true`, then this still gets included in the object here. Should we add a `inherit: undefined` line here after `...layout`?

Reply:
I don't think there's any harm in keeping it. We've been keeping it for the Group block deprecation, and on the whole I think the less we change the less scope there is for things to go wrong 😅

Reply:
The main reason I thought of removing it is that a truthy check for `layout.inherit` is included in the `isEligible` callbacks, so I was wondering if it'd be possible for a migrated block to be treated as though it hasn't been migrated if the inherit value sticks around 🤔

Totally agree with trying to change fewer things, though!

Reply:
We can also change the check to not run at all if `type: "contstrained"` is present.

Reply:
Why other blocks don't have a deprecation except for Query? It's weird to have an empty Query Loop with all the attributes set.. Also the `migrate` function should run only to this deprecation? Usually it's needed to also run for [other migrations](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/)..

Reply:
The Query deprecation doesn't address this change, it should have been added as part of #42763. I added it here in an attempt to fix the issue reported in #43502, as the solution from that PR was causing other breakage.

> Also the migrate function should run only to this deprecation? Usually it's needed to also run for other migrations...

I'm not sure I understand. Is there anything I should change here?

Reply:
I haven't tested, but this is from the [docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/):
>It is important to note that if a deprecation’s save method does not produce a valid block then it is skipped completely, including its migrate method, even if isEligible would return true for the given attributes. This means that if you have several deprecations for a block and want to perform a new migration, like moving content to InnerBlocks, you may need to update the migrate methods in multiple deprecations in order for the required changes to be applied to all previous versions of the block.

You could test by copy/paste the other deprecated versions and see if there are errors.

Reply:
Oh, got it, thanks! I'll have a look.